### ISSUE TITLE
Use input data to generate models / scaffolding

### ISSUE THREAD

**Issue by fleed** [dotnet/templating#394]
**TL;DR**
I'd like to use input data (e.g. a json file) to (re)generate model classes

As developer I've found myself writing similar classes over and over again. Each of us defines some conventions and then writes relative classes in multiple layers (application, data, etc.).  
In the past I've successfully used T4 to generate my classes out of a custom xml file. It works well and I could still use it, but `dotnet new` seems interesting enough to improve my workflow. It is a sort of **continuous scaffolding** that I've also added to my build pipeline (using **cakebuild**) so that whenever I add a property to the model, all templates in all layers are generated again.

Why should I prefer scaffolding to reflection?

- performance
- easier debugging
- errors at compile-time instead of run-time

To accomplish the goal I'd like to define a file containing a custom model definition similar to the following one (just a quick example):

```json
{
    "entities": [
        {
            "name": "Person",
            "properties": [{"name": "Username", "type": "string", "isUnique": true },
                     {"name": "Name", "type": "string" },
                     {"name": "LastName", "type": "string" }]
        }]
}
```

Out of these input data and template files, the `dotnet new` engine should generate the actual classes (entity classes decorated with JSON attributes, data classes decorated with EF attributes, interfaces for repository and data layers, interfaces, etc.). I think that a single file (maybe `scaffold.js`) should contain the logic to iterate through data and invoke the file generation with the correct source template.

This generation should be run multiple times, overriding existing files (if a "force" flag is explicitly specified.. otherwise, it should interactively ask permission to override existing files).  
I'd also like to keep the template sources in my repository side-by-side with code, because the scaffolding templates need to be versioned along with data and code.

**Comment by sayedihashimi** [comment]
Hi @fleed thanks for the comments.

The template engine is really an extensible stream processor. I'm not sure if your proposal fits well with the design but @mlorbetske would know. We have a lot of extensiblity so it may be a good fit. We are having a discussion on extending the Template Engine at https://github.com/dotnet/templating/issues/286. @mlorbetske has already left some comments there. Take a look at the comments he left there. Also I'm sure he'll comment here when he gets a chance.

**Comment by sayedihashimi** [comment]
Hi @fleed, thanks for the issue. We think this sounds interesting, but we have a lot of work that we need to be focused on in the short term.
For the time being we've added this to our list of potential features for the future. Once we make some progress on features like; install/uninstall/update, sharing templates across cli/vs/vs4mac/yeoman, cleaning up help, and a few other important items.

If you're super interested in this and want to take a crack at the code we will review and discuss in a PR.

**Comment by fleed** [comment]
Thank you for attention, @sayedihashimi and @mlorbetske.
I hope that I can contribute to the enhancement.

**Comment by sayedihashimi** [comment]
We are not planning to work on this anytime soon. Closing this for now.
